### PR TITLE
Automatically create release

### PR DESCRIPTION
**Pull request by taooceros**
I copy this workflow from Wox.Plugin.Runner to create automatically create release, so that any merge to main will automatically creates a release based on the version in plugin.json. 
Flow has added new feature to automatically detect update for plugins, so that means we only need to review and merge change in the future.

### CODE REVIEW

**Comment by harrynull** [comment]
Nice! Does it create a new release for every new commits? I wonder if it would be bad user experience if the updates are too frequent (do they get a notification when there is a new version?).
Also, do we still need to change the version number in `plugin.json`? What would happen if there is a new commit that does not change the version number?

**Comment by taooceros** [comment]
> Nice! Does it create a new release for every new commits? I wonder if it would be bad user experience if the updates are too frequent (do they get a notification when there is a new version?).
> Also, do we still need to change the version number in `plugin.json`? What would happen if there is a new commit that does not change the version number?

It only creates a new release for push on main. Generally, user won't have access directly to main, and any commit merged to main should be reviewed. One merge commit will only create one release.
Actually, users won't receive notice when new plugin updates available (although I think they should), so never mind for that.
Therefore, if we mind creating too much release, it would be possible to have a `dev` branch for temp receiving commit, and merge all that to `main` in whole (although I don't think we will have so much pr at a time🤣🤣).
We should always change the version number in plugin.json after every update actually, because that represents our plugin version, even though the one I write for automatically checking update is based on tag (so manually have a higher tag version without changing version in plugin.json will be accepted). 
The tag generated by this workflow is based on version in plugin.json. If we don't change the version number, as what we have tested, it will fail to create new release due to tag duplicate.

**Comment by harrynull** [comment]
I'd create a `release` branch that is specifically for new release and make sure when we merge to that branch the version number is updated. This way we can avoid pushing out a new release for trivial changes like fixing README typo.

**Comment by taooceros** [comment]
> I'd create a `release` branch that is specifically for new release and make sure when we merge to that branch the version number is updated. This way we can avoid pushing out a new release for trivial changes like fixing README typo.

That's a great idea!
As an alternative I have tried, you can add the Readme.md to pathignore so that if only that file has been changed, it won't trigger a new release.
Actually, fail on the action won't have negative effect so even though we forget to change the version number, nothing will be changed.

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: Publish Release
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ main ]
```
Comment (harrynull) [inline .github/workflows/publish.yml:6]:
```suggestion
    branches: [ release ]
```

**Review by harrynull — APPROVED** [review]

**Pull request merged**